spark/src/main/scala/io/delta/sharing/spark/DeltaSharingSourceOffset.scala

```diff
@@ -0,0 +1,150 @@
+/*
+ * Copyright (2021) The Delta Lake Project Authors.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package io.delta.sharing.spark
+
+// scalastyle:off import.ordering.noEmptyLine
+import org.apache.spark.sql.connector.read.streaming.{Offset => OffsetV2}
+import org.apache.spark.sql.execution.streaming.{Offset, SerializedOffset}
+import org.json4s._
+import org.json4s.jackson.JsonMethods.parse
+
+import io.delta.sharing.spark.util.JsonUtils
+
+/**
+ * Tracks how far we processed in when reading changes from the [[Delta Sharing Server]].
+ *
+ * Note this class retains the naming of `Reservoir` to maintain compatibility
+ * with serialized offsets from the beta period.
```

Comment:
Is this applicable to us?

Reply:
good catch, I actually asked @zsxwing about this and it doesn't apply to us, and forgot to update it here.
Changed to tableId and tableVersion